src/coreclr/src/System.Private.CoreLib/src/System/Runtime/InteropServices/Marshal.CoreCLR.cs

```diff
@@ -326,6 +326,16 @@ public static string GetTypeInfoName(ITypeInfo typeInfo)
         /// </summary>
         public static IntPtr /* IUnknown* */ GetIUnknownForObject(object o)
         {
+            if (o is null)
+            {
+                throw new ArgumentNullException(nameof(o));
+            }
+
+            // Passing null as the ComWrapper implementation will use the globally registered wrappper (if available)
+            IntPtr ptrMaybe = ComWrappers.GetOrCreateComInterfaceForObjectInternal(impl: null, o, CreateComInterfaceFlags.None);
```

Comment:
What is the performance impact of doing this on the existing APIs / scenarios? Are there ways to minimize it by factoring the implementation differently?

Reply:
Added a check to avoid calling this if we already know there isn't a registered instance.

In the case of no registered global instance, it was about +100-200ns per call without the check. Not seeing a clear difference with the check.